### PR TITLE
fix(plugins): Allow setting groups with posthog.capture

### PR DESCRIPTION
**Pull request by timgl**
## Problem

posthog.capture inside plugins doesn't work with groups

## Changes

I couldn't find any other tests testing groups, so I have pretty low confidence this is correct

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

???

### CODE REVIEW

File: plugin-server/src/worker/vm/extensions/posthog.ts [hunk 1]
```diff
@@ -41,6 +42,7 @@ async function queueEvent(hub: Hub, pluginConfig: PluginConfig, data: InternalDa
                     now: data.timestamp,
                     sent_at: data.timestamp,
                     uuid: data.uuid,
+                    ...(data.groups ? { groups: data.groups } : {}),
```
Comment (macobo) [inline plugin-server/src/worker/vm/extensions/posthog.ts:45]: The key is incorrect - see https://posthog.com/docs/user-guides/group-analytics for other libraries, the key is $groups not groups.

File: plugin-server/src/worker/vm/extensions/posthog.ts [hunk 2]
```diff
@@ -52,7 +54,12 @@ export function createPosthog(hub: Hub, pluginConfig: PluginConfig): DummyPostHo
 
     return {
         capture: async (event, properties = {}) => {
-            const { timestamp = DateTime.utc().toISO(), distinct_id = distinctId, ...otherProperties } = properties
+            const {
+                timestamp = DateTime.utc().toISO(),
+                distinct_id = distinctId,
+                groups = null,
```
Comment (macobo) [inline plugin-server/src/worker/vm/extensions/posthog.ts:60]: What library are we emulating here? Can we make sure the interface is identical?

File: plugin-server/tests/worker/vm.test.ts [hunk 3]
```diff
@@ -1021,6 +1021,37 @@ describe('vm tests', () => {
         })
     })
 
+    test('posthog.capture accepts groups', async () => {
```
Comment (macobo) [inline plugin-server/tests/worker/vm.test.ts:1024]: Note: We've been trying to move away from integration-only test model. They are hard to debug and often bugs slip through.

With this in mind, I'd leave this test be, but could you add also a unit test file for extensions/posthog.ts with your new test?

File: plugin-server/src/worker/vm/extensions/posthog.ts (hunk 1 above)
Comment (macobo) [inline plugin-server/src/worker/vm/extensions/posthog.ts:45]: This doesn't work - we don't send groups information under top-level of the message. it's supposed to be data.properties.$groups

File: plugin-server/src/worker/vm/extensions/posthog.ts [hunk 4]
```diff
@@ -65,6 +72,7 @@ export function createPosthog(hub: Hub, pluginConfig: PluginConfig): DummyPostHo
                 },
                 team_id: pluginConfig.team_id,
                 uuid: new UUIDT().toString(),
+                ...(groups ? { groups } : {}),
```
Comment (macobo) [inline plugin-server/src/worker/vm/extensions/posthog.ts:75]: Same here:
- The key is `$groups`, see see https://posthog.com/docs/user-guides/group-analytics under other libraries
- Groups should be nested under $properties not at top level

**Comment by timgl** [comment]
Sorry gang, for some reason I thought passing `groups` inside the properties wasn't working, but it is. Ignore!

**Pull request closed without merging**